scripts/integtest.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -e
+
+function usage() {
+    echo ""
+    echo "This script is used to run integration tests for plugin installed on a remote OpenSearch/Dashboards cluster."
```

Comment:
There seem to be quite a few different concepts which are referred to as "integration tests" in this repository.

It might make sense to add a bit of an explanation when this script is used and what makes it different from other test functionality.

Reply:
This is coming from opensearch-build repo: https://github.com/opensearch-project/opensearch-build/blob/a58db5a2f3f8794fb91c7c87cf61b9852b9004e0/DEVELOPER_GUIDE.md?plain=1#L309, which is a way for us to pass a custom integ test script to the build repo when running release time tests with RCs.

Reply:
There are 2 concepts of integTest.

1. `integTest` - This task is project wide and uses testclusters when run as a PR check. When run with release-testing, the build team spins up a cluster with the release candidate and runs these tests against that cluster
2. `integrationTest` - This task is specific to the security plugin and utilizes in-memory nodes/clusters - you introduced the code so I know you are aware ;)

At release-time, only tests in 1) should be run. Currently, the security plugin has a very small number of these tests and you can find them in this package: https://github.com/opensearch-project/security/tree/main/src/test/java/org/opensearch/security/sanity/tests

There should be more of these tests added to ensure that integration tests in various scenarios are run against the RC at release time.

Reply:
I think it would be also possible to build the ability to target an external cluster into the `LocalCluster` framework of the `integrationTest` folder. However, there will be a couple of cases where this won't work - for example cases where special opensearch.yml values are needed or where there are several clusters at once (for CCS tests).

However, I guess this should be discussed in a separate issue, not this PR :)

My point here was just: As the terms `integTest` and `integrationTest` are so similar while referring to a different thing, there should be somewhere a short explanation on the difference to make the difference clear for newcomers to the project. But that's just IMHO :)

Reply:
Agreed. The only reference I can find in markdown is [here](https://github.com/opensearch-project/security/blob/b445f43c4f043da0db47e0609284b214f062bcfc/DEVELOPER_GUIDE.md?plain=1#L270-L272)

Reply:
This [file in the opensearch-plugins repo](https://github.com/opensearch-project/opensearch-plugins/blob/main/TESTING.md?plain=1) also describes testing patterns across plugins and is a good area to add documentation to.